File: Python3/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        reversed_x = 0
        x_copy = x
        while x_copy > 0:
            reversed_x *= 10
            reversed_x += x_copy % 10
            x_copy //= 10

        return x == reversed_x

File: Rust/palindrome-number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let mut x_copy = x;
        let mut x_reversed = 0;

        while x_copy > 0 {
            x_reversed *= 10;
            x_reversed += x_copy % 10;
            x_copy /= 10;
        }
        x_reversed == x
    }
}
